plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSnapshots.java

```diff
@@ -33,7 +37,264 @@
+        assertUpdate(format("INSERT INTO %s VALUES(1, 0)", tableName), 1);
+        assertUpdate(format("INSERT INTO %s VALUES(1, 1), (1, 2), (1, 3), (1, 4)", tableName), 4);
+        // No snapshot generated
+        assertUpdate(format("ALTER TABLE %s SET PROPERTIES format_version = 2", tableName));
+
+        // Row level delete that generates a single MOR file and leaves original file as some rows are still valid.
+        QueryId singleRowDeleteQueryId = getDistributedQueryRunner()
+                .executeWithQueryId(getSession(), format("DELETE FROM %s WHERE a = 1 AND b = 2", tableName))
+                .getQueryId();
+
+        assertThat(getLatestQueryIdFromSnapshotSummary(tableName))
+                .isEqualTo(singleRowDeleteQueryId);
+
+        // Delete that generates MOR files and also removes a file that's no longer present in the latest snapshot
+        // This generates two snapshots as part of a transaction, one for the MOR files and one for cleaning up
+        // the unneeded files.
+        QueryId multiRowDeleteWithFileCleanUpQueryId = getDistributedQueryRunner()
+                .executeWithQueryId(getSession(), format("DELETE FROM %s WHERE a = 1", tableName))
+                .getQueryId();
+
+        // Ensure two snapshots from one transaction are generated, one for row deltas and one for removing the empty
+        // files (BaseRowDelta and StreamingDelete)
+        assertThat(getQueryIdsFromSnapshotSummariesByCreationOrder(tableName))
+                .endsWith(multiRowDeleteWithFileCleanUpQueryId, multiRowDeleteWithFileCleanUpQueryId)
+                .size().isEqualTo(6);
+    }
+
+    @Test
+    public void testExpireSnapshotsDoesNotGenerateNewSnapshot()
+    {
+        final String tableName = "test_row_level_delete_summary_query_id" + randomTableSuffix();
+        assertUpdate(format("CREATE TABLE %s (a bigint, b bigint)", tableName));
+        assertUpdate(format("INSERT INTO %s VALUES(1, 0)", tableName), 1);
+
+        QueryId latestSnapshotQueryId = getDistributedQueryRunner()
+                .executeWithQueryId(getSession(), format("INSERT INTO %s VALUES(1, 1), (1, 2), (1, 3), (1, 4)", tableName))
+                .getQueryId();
+
+        assertThat(getSnapshotsIdsByCreationOrder(tableName))
+                .size().isEqualTo(3);
+
+        assertThat(getQueryIdsFromSnapshotSummariesByCreationOrder(tableName))
+                .endsWith(latestSnapshotQueryId);
+
+        assertUpdate(format("ALTER TABLE %s EXECUTE EXPIRE_SNAPSHOTS (retention_threshold => '0s')", tableName));
+
+        // Ensure there's only one snapshot remaining
+        assertThat(getSnapshotsIdsByCreationOrder(tableName))
+                .size().isEqualTo(1);
+
+        // Ensure the remaining snapshot is the same snapshot as the last insert
+        assertThat(getQueryIdsFromSnapshotSummariesByCreationOrder(tableName))
+                .containsExactly(latestSnapshotQueryId)
+                .size().isEqualTo(1);
+    }
```

Comment:
This test serves as ok documentation of the fact that this does not generate a snapshot summary, but it feels like overkill.

It does test that snapshots are expired properly though, and this is `TestIcebergSnapshots`. If we're not testing this in another place currently, we should consider keeping it or adding something similar in a follow up PR.